### PR TITLE
Prevent the same orphan from getting enqueued twice

### PR DESCRIPTION
**Pull request by ptschip**

### CODE REVIEW

File: src/txadmission.cpp [hunk 1]
```diff
@@ -1038,39 +1038,46 @@ void ProcessOrphans(std::vector<uint256> &vWorkQueue)
...
         WRITELOCK(orphanpool.cs);
-        for (auto hash : vEraseQueue)
-            orphanpool.EraseOrphanTx(hash);
-        //  BU: Xtreme thinblocks - purge orphans that are too old
+        for (auto &it : mapEnqueue)
+        {
+            // If the orphan was not erased then it must already have been erased/enqueue'd by another thread
+            // so do not enqueue this orphan again.
+            if (!orphanpool.EraseOrphanTx(it.first))
+                mapEnqueue.erase(it.first);
```
Comment (gandrewstone) [inline src/txadmission.cpp]: it = mapEnqueue.erase(it.first);

(can't erase what's under an iterator and then expect to continue on)

File: src/txorphanpool.h [hunk 2]
```diff
@@ -46,7 +46,7 @@ class CTxOrphanPool
     bool AddOrphanTx(const CTransactionRef &ptx, NodeId peer);
 
     //! Erase an ophan tx from the orphan pool
-    void EraseOrphanTx(uint256 hash);
```
Comment (gandrewstone) [inline src/txorphanpool.h:49]: Please doc-comment the return value

Comment (ptschip) [reply src/txorphanpool.h:49]: done

File: src/txadmission.cpp (hunk 1 above)
Comment (ptschip) [reply src/txadmission.cpp]: done

**Pull request merged**